### PR TITLE
Update Kotlin example with helper methods

### PR DESCRIPTION
**Pull request by jaguililla**
- Upgrade example to the latests Ratpack and Kotlin versions
- Add helper methods to take advantage of Kotlin's syntax
- Refactor example and tests

@alkemist @robfletcher @zedar @erdi @rhart Could you please review? If something is wrong, I've got no problem in changing/fixing it

### CODE REVIEW

**Comment by rhart** [comment]
Great stuff! Thanks, @jamming!

If no one has got to this before I get back from hols I'll merge on my return.

**Comment by jaguililla** [comment]
Thanks!

File: src/main/kotlin/ratpack/example/kotlin/Helpers.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+@file:Suppress("Unused")
+
+/*
+ * Helper functions to make the example more Kotlin like. All of them are extension functions that add
+ * syntactic sugar to Ratpack's methods.
+ *
+ * The names are prefixed by 'k' because the signature of some methods is the same that Ratpack's ones and
```
Comment (rhart) [inline src/main/kotlin/ratpack/example/kotlin/Helpers.kt]: I'm not a Kotlin expert but I'm not a huge fan of 'k' prefix.  What you're demonstrating is great, effectively a Kotlin DSL for Ratpack, but I wonder whether it belongs in this example.  If you were going to use Kotlin with Ratpack today, is this what you would do idiomatically?

@robfletcher do you have an opinion here?

File: src/main/kotlin/ratpack/example/kotlin/Main.kt [hunk 2]
```diff
@@ -1,61 +1,57 @@
...
+        // Set up a nested routing block, which is delegated to `nestedHandler`
+        kprefix("nested") {
+            kpath(":var1/:var2?") {
+                // The path tokens are the :var1 and :var2 path components above
+                render("from the nested handler, var1: ${pathTokens["var1"]}, var2: ${pathTokens["var2"]}")
+            }
+        }
+
+        // Map to a dependency injected handler
+        path("injected", MyHandler::class.java)
```
Comment (rhart) [inline src/main/kotlin/ratpack/example/kotlin/Main.kt]: I think I would prefix this with 'k' to keep it consistent.

File: src/main/kotlin/ratpack/example/kotlin/MyModule.kt [hunk 3]
```diff
@@ -8,18 +8,21 @@ import ratpack.handling.HandlerDecorator
...
+     *
+     * @see MyHandler
+     */
+    override fun configure() {
+        bind(MyService::class.java).to(MyServiceImpl::class.java)
+        bind(MyHandler::class.java)
+        Multibinder.newSetBinder(binder(), HandlerDecorator::class.java)
+            .addBinding()
+            .toInstance(HandlerDecorator.prepend {
+                println("Received: ${it.request.uri}")
```
Comment (rhart) [inline src/main/kotlin/ratpack/example/kotlin/MyModule.kt]: I would keep `LoggingHandler`, it's demonstrating a handler implicitly set up by a module

File: src/test/kotlin/ratpack/example/kotlin/SiteTest.kt [hunk 4]
```diff
@@ -1,67 +1,54 @@
...
-import ratpack.test.MainClassApplicationUnderTest
-import java.io.InputStreamReader
-import kotlin.test.assertEquals
-import kotlin.test.fail
 
 @RunWith(JUnit4::class)
 class SiteTest {
+    val app = kappFromServer(createServer ())
 
-  val aut = MainClassApplicationUnderTest(Main::class.java)
```
Comment (rhart) [inline src/test/kotlin/ratpack/example/kotlin/SiteTest.kt:15]: I would still use `MainClassApplicationUnderTest` for testing.  It has functionality like Impositions that you will need for "real" testing.

**Comment by rhart** [comment]
@jamming I've added some comments

**Comment by jaguililla** [comment]
@rhart @robfletcher 

I've made all PRs proposal, except for the use of `MainClassApplicationUnderTest` because it requires an object with a main method and it is less "Kotlin idiomatic".

By the way, I think your project is awesome and I'm using it with Kotlin. If you agree, I can make a proper integration and submit the PR.

Thanks for your feedback!

**Comment by rhart** [comment]
@jamming I think `MainClassApplicationUnderTest` is a must have change I'm afraid.  It adds additional functionality you will defo need for testing in a real app.  It might be less "Kotlin idiomatic" but is certainly Ratpack idiomatic which trumps it :)

I saw the issue you raised around adding a Kotlin DSL.  Will leave Luke to comment.

Thanks again for the PR!

**Comment by jaguililla** [comment]
Done!

**Comment by rhart** [comment]
Many thanks @jamming, great stuff!

**Pull request merged**